demo/admin/src/products/future/generated/ProductsGrid.tsx

```diff
@@ -124,6 +124,17 @@ export function ProductsGrid({ filter, toolbarAction, rowAction }: Props): React
                 { value: "Shirt", label: intl.formatMessage({ id: "product.type.shirt", defaultMessage: "Shirt" }) },
                 { value: "Tie", label: intl.formatMessage({ id: "product.type.tie", defaultMessage: "Tie" }) },
             ],
+            renderCell: ({ row, colDef }) => {
+                if (colDef.valueOptions && Array.isArray(colDef.valueOptions)) {
+                    const selectedOption = colDef.valueOptions.find((option) => typeof option === "object" && option.value === row.type);
+
+                    if (selectedOption && typeof selectedOption === "object") {
+                        return selectedOption.label;
+                    }
+                }
```

Comment:
this is a general valid code for all staticSelects and could come from the library

Reply:
I'd leave it as it is for now, as this will likely be replaced in #2422, depending on the outcome of https://github.com/vivid-planet/comet/pull/2422#discussion_r1724624780. 
We can add a function to the library when we know the final implementation.